EmailAddressPolicy: 'The operation can't be performed on the default email address policy'

### Description of the issue

When trying to set a specific custom domain ({ExoDomain} via this code block, we are met with the error below:

` EmailAddressPolicy             = @(
                @{
                    UniqueId                          = 'Default Policy'
                    EnabledEmailAddressTemplates      = @('SMTP:@{{ExoDomain}}')
                    EnabledPrimarySMTPAddressTemplate = @('@{{ExoDomain}}')
                    #Ensure                            = 'Present'
                    ManagedByFilter                   = ''
                    Name                              = 'Default Policy'
                    Priority                          = 'Lowest'
                }`

`The operation can't be performed on the default email address policy`

According to this article, it can be done by omitting certain parameters:
https://rakhesh.com/exchange/the-operation-cant-be-performed-on-the-default-email-policy/

However, when omitting e.g. the 'Name' attribute, we are met with:

`##[error]     [-] NonNodeData.Exchange.EmailAddressPolicy 2ms (2ms|0ms)
##[error]      Expected a value, because Required setting, but got $null or empty.
##[error]      at $inputObject.NonNodeData.Exchange.EmailAddressPolicy[0].Name | Should -Not -BeNullOrEmpty -Because 'Required setting', C:\Users\AzDevOps\AppData\Local\Temp\tmp3zxbdk.tests.ps1:606`

Apparantly, there is a bug in this part of the resource:

https://github.com/Microsoft365DSC/Microsoft365DSC/blob/614f477acff8f0898c58e262e9f90f23870050a6/Modules/Microsoft365DSC/DSCResources/MSFT_EXOEmailAddressPolicy/MSFT_EXOEmailAddressPolicy.psm1#L264-L273

The $Identity variable doesn't exist.

### Microsoft 365 DSC Version

PRD / 1.26.617.1

### Which workloads are affected

Exchange Online

### The DSC configuration

```powershell
EmailAddressPolicy             = @(
                @{
                    UniqueId                          = 'Default Policy'
                    EnabledEmailAddressTemplates      = @('SMTP:@{{ExoDomain}}')
                    EnabledPrimarySMTPAddressTemplate = @('@{{ExoDomain}}')
                    #Ensure                            = 'Present'
                    ManagedByFilter                   = ''
                    Name                              = 'Default Policy'
                    Priority                          = 'Lowest'
                }
```

### Verbose logs showing the problem

```powershell

```

### Environment Information + PowerShell Version

```powershell

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,12 @@
 
 # UNRELEASED
 
+* AADCrossTenantAccessPolicyConfigurationPartner
+  * Fixed an issue when updating the policy with invalid property names.
+    FIXES [#7272](https://github.com/Microsoft365DSC/Microsoft365DSC/issues/7272)
+* EXOEmailAddressPolicy
+  * Fixed an issue when updating the default policy.
+    FIXES [#7271](https://github.com/Microsoft365DSC/Microsoft365DSC/issues/7271)
 * EXOOrganizationConfig
   * Fixed an issue where processing would fail if any of the properties
     `DelayedDelicensingEnabledState`, `EndUserMailNotificationForDelayedDelicensingState`

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_AADCrossTenantAccessPolicyConfigurationPartner/MSFT_AADCrossTenantAccessPolicyConfigurationPartner.psm1
@@ -328,7 +328,12 @@ function Set-TargetResource
         $OperationParams.Remove('IdentitySynchronization') | Out-Null
     }
 
-    $OperationParams = Rename-M365DSCCimInstanceParameter -Properties $OperationParams
+    $OperationParams = Rename-M365DSCCimInstanceParameter -Properties $OperationParams -KeyMapping @{
+        B2BCollaborationInbound  = 'b2bCollaborationInbound'
+        B2BCollaborationOutbound = 'b2bCollaborationOutbound'
+        B2BDirectConnectInbound  = 'b2bDirectConnectInbound'
+        B2BDirectConnectOutbound = 'b2bDirectConnectOutbound'
+    }
 
     if ($Ensure -eq 'Present' -and $currentInstance.Ensure -eq 'Absent')
     {

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOEmailAddressPolicy/MSFT_EXOEmailAddressPolicy.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOEmailAddressPolicy/MSFT_EXOEmailAddressPolicy.psm1
@@ -261,16 +261,12 @@ function Set-TargetResource
     # CASE: Email Address Policy exists and it should, but has different values than the desired ones
     elseif ($Ensure -eq 'Present' -and $currentEmailAddressPolicyConfig.Ensure -eq 'Present')
     {
-        if ($Identity -ne 'Default Policy')
+        Write-Verbose -Message "Email Address Policy '$($Name)' already exists, but needs updating."
+        if ($Name -eq 'Default Policy')
         {
-            Write-Verbose -Message "Email Address Policy '$($Name)' already exists, but needs updating."
-            Write-Verbose -Message "Setting Email Address Policy $($Name) with values: $(Convert-M365DscHashtableToString -Hashtable $SetEmailAddressPolicyParams)"
-            Set-EmailAddressPolicy @SetEmailAddressPolicyParams
+            $SetEmailAddressPolicyParams.Remove('Priority')
         }
-        else
-        {
-            Write-Verbose -Message 'Cannot update the Default Email Address Policy.'
-        }
+        Set-EmailAddressPolicy @SetEmailAddressPolicyParams
     }
 }
 

